oraclesrv/views.py

```diff
@@ -22,17 +22,17 @@ def return_response(results, status_code):
     r.headers['content-type'] = 'application/json'
     return r
 
-def get_user_info_from_adsws(session):
+def get_user_info_from_adsws_having_token(session, token):
     """
 
     :param session:
     :return:
     """
-    if session:
+    if session and token:
```

Comment:
should be:

if reader:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + reader
elif token:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + token
elif uid:
     url = current_app.config['ORACLE_SERVICE_ACCOUNT_INFO_URL'] + '/' + uid
elif session:
     ....

Reply:
sorry, I don't understand the first line. when we have the reader, we are all set, right, why should we call adsws now?